### PR TITLE
Adding Examples for Feature Selection Module

### PR DESCRIPTION
**Pull request by datacubeR**
Hi @solegalli ,

Happy New Year, please review the added examples for all the Feature Selection Methods.

PS: Would you check why tests are failing, I think it is not related to my code.

### CODE REVIEW

**Review by solegalli — COMMENTED** [review]
Hey @datacubeR 

Nice to have you back! Thanks for the amount of work in this PR

The tests are failing thanks to sklearn latest release :p 

I am fixing them in this PR #575 

I can finish and merge by tomorrow evening (Buenos Aires time) and then, if you rebase onto your branch, the tests should pass.

Also, I made some minor suggestions to try and give users the full flavour of these transformers. Could you please look at those while I fix the tests?

Thanks a lot!

**Comment by solegalli** [comment]
Hey @datacubeR 

tests have now been fixed in #583 

If you rebase main onto your feature-branch, that should do it.

File: feature_engine/selection/drop_correlated_features.py [hunk 1]
```diff
@@ -109,6 +109,22 @@ class DropCorrelatedFeatures(BaseSelector):
...
     feature_engine.selection.SmartCorrelationSelection
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.selection import DropCorrelatedFeatures
+    >>> X = pd.DataFrame(dict(x1 = [1,2,1,1],
+    >>>                     x2 = [2,4,3,1],
+    >>>                     x3 = [True, False, False, False]))
```
[the remark this answers is not in the document]
Comment (datacubeR) [reply feature_engine/selection/drop_correlated_features.py:120]: I'll use only numerical

File: feature_engine/selection/information_value.py [hunk 2]
```diff
@@ -138,6 +138,25 @@ class SelectByInformationValue(BaseSelector, WoE):
 
     .. [2] WoE and IV for continuous variables
         https://www.listendata.com/2019/08/WOE-IV-Continuous-Dependent.html
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.selection import SelectByInformationValue
```
[the remark this answers is not in the document]
Comment (datacubeR) [reply feature_engine/selection/information_value.py:146]: Changed x3 to strings...

File: feature_engine/selection/smart_correlation_selection.py [hunk 3]
```diff
@@ -149,6 +149,35 @@ class SmartCorrelatedSelection(BaseSelector):
...
     feature_engine.selection.DropCorrelatedFeatures
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.selection import SmartCorrelatedSelection
+    >>> X = pd.DataFrame(dict(x1 = [1,2,1,1],
+    >>>                     x2 = [2,4,3,1],
+    >>>                     x3 = [True, False, False, False]))
```
[the remark this answers is not in the document]
Comment (datacubeR) [reply feature_engine/selection/smart_correlation_selection.py:160]: Changed to numeric

File: feature_engine/selection/smart_correlation_selection.py [hunk 4]
```diff
@@ -149,6 +149,35 @@ class SmartCorrelatedSelection(BaseSelector):
...
+    0   1   True
+    1   2  False
+    2   1  False
+    3   1  False
+
+    It is also possible alternative selection methods, in this case seleting features with higher variance:
+
+    >>> X = pd.DataFrame(dict(x1 = [1000,2000,1000,1000],
+    >>>                     x2 = [2,4,3,1],
+    >>>                     x3 = [True, False, False, False]))
```
[the remark this answers is not in the document]
Comment (datacubeR) [reply feature_engine/selection/smart_correlation_selection.py:173]: Same

File: feature_engine/selection/drop_psi_features.py [hunk 5]
```diff
@@ -210,6 +210,22 @@ class DropHighPSIFeatures(BaseSelector):
     .. [1] Yurdakul B. "Statistical properties of population stability index".
        Western Michigan University, 2018.
        https://scholarworks.wmich.edu/dissertations/3208/
+
+    Examples
+    --------
+
+    >>> import pandas as pd
```
[the remark this answers is not in the document]
Comment (datacubeR) [reply feature_engine/selection/drop_psi_features.py:217]: Had to create a longer dataset since it needs at least 10 elements by fold to work.

**Comment by datacubeR** [comment]
Hi @solegalli. 
Sorry for the delay but I fixed all the suggestions given above and pass all the tests. 
Let me know if everything is OK.

Thanks.

Alfonso

**Comment by solegalli** [comment]
Hey @datacubeR 

For some reason I didn't notice that you'd already made the requested changes. I am so sorry. I just noticed today. Anyhow, I am merging as we speak! Thank you so much for this!

**Pull request merged**